### PR TITLE
Updated notes on scalability of various estimators

### LINKED ISSUE #109: How does dask-ml handle large datasets?
From the latest demos of dask-sklearn (and related projects) that I've seen, dask has primarily focused on parallelizing grid/random search and CV, and provides some smart caching mechanisms for transformers and estimators that have already been evaluated on a dataset.

However, that approach breaks down when, say, a slow algorithm is being trained on a very large dataset. With some slower algorithms, training one instance of the algorithm on one fold of the dataset can take hours or days, which makes training the algorithm unviable for all but the most patient practitioner.

Can dask-ml speed up the underlying ML algorithms that it supports?

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes #109

### CODE REVIEW

**Comment by TomAugspurger** [comment]
I'm not sure if this is the best approach. The short version is that

1. All the estimators in dask-ml will work in parallel on distributed arrays.
2. The `Incremental*` wrappers will work in serial over the blocks, so they won't benefit from any parallelism, beyond what scikit-learn implements
3. The meta-estimators like `RandomizedSearchCV` inherit whatever parallelism the actual estimator implements. If you search over `dask_ml.linear_model.LogisticRegression`, things should be fine on a large array. If you search over `sklearn.linear_model.LogisticRegression` not so much.

We should probably have something like that in `index.rst` but I don't want to over-complicate things too early.

Adding more scalability examples as suggested by @marcelsan would be great (though we'll want to balance this against the time it takes to do a full doc build).

cc @rhiver

**Comment by rhiever** [comment]
Thank you for the clarification and I think this is a good addition to the docs. Are there plans to implement more estimators in dask-ml in the future?

File: docs/source/modules/api.rst [hunk 1]
```diff
@@ -43,6 +47,12 @@ Incremental Learning
 
 .. currentmodule:: dask_ml
 
+Some scikit-learn estimators support out-of-core training through the
+``partial_fit`` method. The following estimators wrap those scikit-learn
+estimators, allowing them to be used in Pipelines and on Dask arrays and
+dataframes. Training will still be serial, so these will not benefit from
+a paralell or distributed training any more than the underlying estimator.
```
Comment (rhiever) [inline docs/source/modules/api.rst]: typo: paralell --> parallel

**Comment by TomAugspurger** [comment]
> Are there plans to implement more estimators in dask-ml in the future?

Yep, as driven by user / developer interest.

**Pull request merged**